apis/s3/v1beta1/lifecycleConfiguration_types.go

```diff
@@ -46,7 +46,7 @@ type LifecycleRule struct {
 	Filter *LifecycleRuleFilter `json:"filter,omitempty"`
 
 	// Unique identifier for the rule. The value cannot be longer than 255 characters.
-	ID *string `json:"ID,omitempty"`
+	ID *string `json:"id,omitempty"`
```

Comment:
@muvaf this would be a breaking change and I think would necessitate a minor release, rather than a patch.

Reply:
@hasheddan Do you think we should bump it to `v1beta2`? Also, I can separate that commit from the others and save jsontag changes for the major release.

Reply:
Separating commits sounds good :) And I think the correct versioning would be to bump to `v1beta2`. I wonder if this would be a good opportunity to experiment with conversion webhooks...

Reply:
I think we can get away with providing manual instructions for now per [Kubernetes versioning](https://github.com/kubernetes/community/blob/3302b37/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions).

Reply:
So I went ahead and removed the actual breaking changes and the diff is as following:
<img width="939" alt="image" src="https://user-images.githubusercontent.com/7584126/96333629-c1954200-1073-11eb-8ba1-41a5757a39a0.png">

However, I realized that the sub-resources that those fields belong didn't really work in the last release anyway. So, while technically it's more correct to bump to `v1beta2`, in practice I feel like we won't be breaking anyone using `Bucket` resource if we changed those fields and did a patch release. @hasheddan what do you think?

Reply:
@hasheddan pinging

Reply:
Hmm I am not entirely sure the current behavior of these subresources, my only reservation would be if this could cause a case where a currently valid S3 bucket instance now failed (i.e. if we change field names that are required then a valid instance would become invalid). Since we don't have webhooks yet, maybe we should keep as `v1beta1` but just do a `v0.14` release?

Reply:
> if this could cause a case where a currently valid S3 bucket instance now failed

A valid working bucket wouldn't fail because in order to be working, it shouldn't include those fields anyway. If it did, then it's failing because of the errors fixed.

Reply:
@hasheddan Oh by the way, a full release sounds good.